### PR TITLE
fast forward master

### PR DESCRIPTION
**Pull request by rhaschke**
- fix `Task::operator=(Task&&)`
- ~~merge of python API~~

### CODE REVIEW

File: core/python/wrapper/src/core.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	return l;
+}
+
+void Stage_setForwardedProperties(Stage& self, const bp::list& names) {
+	boost::python::stl_input_iterator<std::string> begin(names), end;
+	self.setForwardedProperties(std::set<std::string>(begin, end));
+}
+
+
+void ContainerBase_insert(ContainerBase& self, std::auto_ptr<Stage> stage, int before = -1) {
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply core/python/wrapper/src/core.cpp]: I know, but there is no other way to convince boost::python to handle the std::unique_ptr used for MTC stages. That's one reason, why we want to switch to std::shared_ptrs in MTC.

File: core/python/wrapper/src/core.cpp [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+#include <moveit_task_constructor_msgs/Solution.h>
+
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+
+namespace bp = boost::python;
+using namespace moveit::task_constructor;
+
+namespace moveit {
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply core/python/wrapper/src/core.cpp]: Not, if I can decide ;-)
Introducing a clang-format file make sense though.

File: core/python/src/moveit/task_constructor/core.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from _core import *
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply core/python/src/moveit/task_constructor/core.py]: Currently, these files are trivial. But in future, we might want to add python-only code here.

**Comment by rhaschke** [comment]
@henningkayser, Michael asked me to merge my python branch into the master (although I still consider it work in progress and didn't like the idea very much). 
Anyway, this cannot be split into smaller pieces - it's all about the python extensions.

**Comment by rhaschke** [comment]
Due to the incompleteness of and warnings in the python API as well as due to complaints of @henningkayser, I decided to hold back the python merge for now.

**Comment by v4hn** [comment]
> Due to the incompleteness of and warnings in the python API as well as due to complaints of @henningkayser

Thank you so much @henningkayser :stuck_out_tongue: .
Feel free to improve the current state of the python-branch then.

I would like to keep the WIP python code-base in the upstream repository at least, so that people can find it here. Personally, I wouldn't complain about incomplete python-API in upstream - That's the state we have in MoveIt for ages.... :sigh:
Maybe you could push it to a `wip-python-api` branch for now @rhaschke ?

**Comment by rhaschke** [comment]
> Maybe you could push it to a wip-python-api branch for now.

Done: #99

**Comment by rhaschke** [comment]
@v4hn, could you please review and fast-forward this PR. I think, now it has become really simple :wink:

File: core/include/moveit/task_constructor/stage_p.h [hunk 4]
```diff
@@ -124,7 +125,7 @@ class StagePrivate {
 	bool storeFailures() const { return introspection_ != nullptr; }
 
 protected:
-	Stage* const me_; // associated/owning Stage instance
+	Stage* me_; // associated/owning Stage instance
```
Comment (rhaschke) [inline core/include/moveit/task_constructor/stage_p.h]: constness forbids `operator=`.

**Comment by v4hn** [comment]
> Done: #99

Thanks!

> @v4hn, could you please review and fast-forward this PR. I think, now it has become really simple wink

You beat me to it, I just finished looking through the changes.
I wouldn't call adding a pimpl-layer to the core class and implementing a hierarchical swap for everything juggling pointers a "really simple change" :1st_place_medal:

**Comment by rhaschke** [comment]
Yeah, it was some effort to cleanly implement the move assignment operator. Using a hidden impl pointer was the easiest solution ;-)

**Comment by v4hn** [comment]
> Yeah, it was some effort to cleanly implement the move assignment operator.

Thank you so much though!

**Pull request merged**